### PR TITLE
Adding fallback to win32 if onecore is missing

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

**Comment by moozzyk** [comment]
@davidfowl  - this is what I am proposing (not tested)

File: src/dnx.windows/dnx.cpp [hunk 1]
```diff
@@ -3,6 +3,18 @@
 
 #include "stdafx.h"
 
+HMODULE load_dnx_dll(const wchar_t* dnx_dll_name)
+{
+    auto dnx_dll = LoadLibraryEx(dnx_dll_name, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
+    if (!dnx_dll)
+    {
+        auto last_error = GetLastError();
+        _tprintf(L"%s could not be loaded. Last error: %d\n", L"dnx.onecore.dll", last_error);
```
Comment (davidfowl) [inline src/dnx.windows/dnx.cpp]: Use dnx_dll_name here

**Comment by davidfowl** [comment]
I'm going to test it out cuz I needs it :smile:

File: src/dnx.windows/dnx.cpp [hunk 2]
```diff
@@ -15,13 +27,20 @@ int _tmain(int argc, _TCHAR* argv[])
 
     bool is_oneCore = version_info.dwMajorVersion >= 10;
 
-    auto dnx_dll_name = is_oneCore ? L"dnx.onecore.dll" : L"dnx.win32.dll";
+    HMODULE dnx_dll = nullptr;
+
+    if (is_oneCore)
+    {
+        dnx_dll = load_dnx_dll(L"dnx.win32.dll");
```
Comment (davidfowl) [inline src/dnx.windows/dnx.cpp]: should be onecore

File: src/dnx.windows/dnx.cpp (hunk 1 above)
Comment (moozzyk) [inline src/dnx.windows/dnx.cpp]: Yes.

File: src/dnx.windows/dnx.cpp (hunk 2 above)
Comment (moozzyk) [inline src/dnx.windows/dnx.cpp]: Yes. Just noticed that. (Too hasty)

File: src/dnx.common/tpa.cpp [hunk 3]
```diff
@@ -23,6 +23,7 @@ const std::vector<const dnx::char_t*> CreateTpaBase(bool native_images)
             _X("System.Collections.ni.dll"),
             _X("System.Collections.Concurrent.ni.dll"),
             _X("System.ComponentModel.ni.dll"),
+            _X("System.Console.ni.dll"),
```
Comment (davidfowl) [inline src/dnx.common/tpa.cpp]: Don't check this in. I removed it in another commit. Something strange going on or we have different CoreCLR builds... Or something else is going on..

Comment (davidfowl) [inline src/dnx.common/tpa.cpp]: See https://github.com/aspnet/dnx/commit/b2445980bc5528bd066a260fccc2158d5f8406f3

Comment (moozzyk) [inline src/dnx.common/tpa.cpp]: OK.

**Comment by moozzyk** [comment]
@davidfowl  Just tested. I am going to merge this to dev (after reverting changes to tpa)

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**